src/cryptonote_core/tx_verification_utils.cpp

```diff
@@ -422,6 +441,22 @@ static bool collect_fcmp_pp_tx_verify_inputs(cryptonote::transaction &tx,
 namespace cryptonote
 {
 
+bool collect_pubkeys_and_commitments(const transaction& tx, std::vector<rct::key> &pubkeys_and_commitments_inout)
+{
+    for (std::size_t i = 0; i < tx.vout.size(); ++i)
+    {
+        crypto::public_key output_pubkey;
+        if (!cryptonote::get_output_public_key(tx.vout[i], output_pubkey))
+            return false;
+
+        rct::key pubkey = rct::pk2rct(output_pubkey);
+        pubkeys_and_commitments_inout.emplace_back(std::move(pubkey));
+        pubkeys_and_commitments_inout.emplace_back(rct::getCommitment(tx, i));
```

Comment:
The function `rct::getCommitment` can throw if the transaction is improperly formatted, but this function is supposed to return `false` on failure, not throw. Since it is used in a consensus context, we don't want it to be throwing when it should be returning `false`.